packages/core/src/components/tabs/tabs.tsx

```diff
@@ -141,15 +163,63 @@ export class Tabs extends AbstractPureComponent<ITabsProps, ITabsState> {
                     role="tablist"
                 >
                     {tabIndicator}
-                    {tabTitles}
+                    {this.props.overflow ?
+                         <OverflowList
+                            items={tabTitles}
+                            collapseFrom="end"
+                            observeParents={true}
+                            overflowRenderer={this.overflowRenderer}
+                            visibleItemRenderer={this.visibleItemsRenderer}
+                            ref={this.refHandlers.overflowList}
+                            {...this.props.overflowListProps}
+                        /> : tabTitles
+                    }
                 </div>
                 {tabPanels}
             </div>
         );
     }
 
+    public overflowRenderer(items: Array<React.Component>) {
+        if (this.props.overflowRenderer) {
+            return this.props.overflowRenderer(items);
+        }
+
+        const overflowMenu = (
+            <Menu>
+                {items.map((item) => {
+                    if (item.props.panel) {
+                        return (
+                            <MenuItem
+                                text={item.props.title}
+                                onClick={(e) => item.props.onClick(item.props.id, e)}
+                                disabled={item.props.disabled}
+                                active={item.props.selected}
+                            />
+                        );
+                    } else {
+                        return item;
+                    }
+                })}
+            </Menu>
+        );
+
+        return (
+            <Popover content={overflowMenu}>
+                <span style={{ lineHeight: "30px", cursor: "pointer" }}>Overflow</span>
```

Comment:
Make this customizable, default to something more obvious like "+{numberOfHiddenTabs}" maybe @giladgray?